src/components/change-password/change-password.tsx

```diff
@@ -0,0 +1,148 @@
+    ItemIdsOrdered:null,
+    Content:[ {
+        Type:RestSdkTypes.Pages,
+        Variations:null
+    }]
+};
+
+export async function ChangePassword(props: WidgetContext<ChangePasswordEntity>) {
+    const entity = {
+        PostPasswordChangeRedirectPage: defaultMixedContent,
+        PostPasswordChangeMessage:  'Your password was changed successfully!',
+        Header: 'Change password',
+        CurrentPassword: 'Current password',
+        NewPassword: 'New password',
+        ConfirmPassword: 'Repeat new password',
+        SubmitButtonLabel: 'Save',
+        LoginFirstMessage: 'You need to be logged in to change your password.',
+        ValidationRequiredMessage: 'All fields are required.',
+        ValidationMismatchMessage: 'New password and repeat password don\'t match.',
+        ExternalProviderMessageFormat: 'Your profile does not store any passwords, because you are registered with ',
+        ...props.model.Properties
+    };
+    const context = props.requestContext;
+    const dataAttributes = htmlAttributes(props);
+    const defaultClass =  entity.CssClass;
+    const marginClass = entity.Margins && StyleGenerator.getMarginClasses(entity.Margins);
+    const webServicePath =  process.env.SF_WEB_SERVICE_PATH;
+
+    dataAttributes['className'] = classNames(
+        defaultClass,
+        marginClass
+        );
+    dataAttributes['data-sfemptyiconaction'] = 'Edit';
+    dataAttributes['data-sfhasquickeditoperation'] = true;
+
+    const viewModel: any = {
+        ChangePasswordHandlerPath: `${webServicePath}/ChangePassword`,
+        Attributes: entity.Attributes,
+        Labels: {}
+    };
+
+    viewModel.Labels.Header = entity.Header;
+    viewModel.Labels.OldPassword = entity.CurrentPassword;
+    viewModel.Labels.NewPassword = entity.NewPassword;
+    viewModel.Labels.RepeatPassword = entity.ConfirmPassword;
+    viewModel.Labels.SubmitButtonLabel = entity.SubmitButtonLabel;
+    viewModel.Labels.LoginFirstMessage = entity.LoginFirstMessage;
+    viewModel.Labels.ValidationRequiredMessage = entity.ValidationRequiredMessage;
+    viewModel.Labels.ValidationMismatchMessage = entity.ValidationMismatchMessage;
+    viewModel.Labels.ExternalProviderMessageFormat = entity.ExternalProviderMessageFormat;
+    viewModel.VisibilityClasses = StylingConfig.VisibilityClasses;
+    viewModel.InvalidClass = StylingConfig.InvalidClass;
+
+
+    viewModel.PostPasswordChangeAction = entity.PostPasswordChangeAction;
+    let baseURL;
+    if (process.env.NODE_ENV === 'development'){
+        process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+        baseURL = `${process.env.NEXT_PUBLIC_CMS_URL}sf/system/users/current`;
+    }
+    const argsLocal = {
+        BaseURL: baseURL,
+        Name: 'users/current',
+        AdditionalHeaders: {
+            cookie : context.cookie || ''
+        }
+    };
+    const response: any = await RestService.getUnboundType(argsLocal);
+    const user: any = response.value;

```

Comment:
[Suggestion]: see if there is an interface for user

Reply:
aded the interface